### PR TITLE
Removed gradients from isis

### PR DESCRIPTION
**Pull request by ghost**
Gradients are - as we all know - behind us, in the past.

So why are we still using them in our flagship admin template?
Let's get them out!
gradients

1) This PR **removes the gradients** from the login screen (background and box), admin menu and tool bars and the side menu.

2) It also **removes the "Sidebar" title** from the sidebar and tweaks the sidebar styling a bit to make it look better with the no-gradient styling and no title.

3) Thirdly it **adds minified css files** for the isis template and uses them. Also the less files are cleaned up a bit and the css files have been re-compiled.
### Performance gain

The removal of the gradients/shadows are not only a visual improvement (for this day and age), but also provides a significant performance gain. With rendering speeds increasing around the 20% mark!
The minified css files will increase the page load speed even more.

### CODE REVIEW

**Comment by brianteeman** [comment]
Can I request that we leave the isis template alone and concentrate effort on a new template instead<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6340">issues.joomla.org/joomla-cms/6340</a>.</sub>

File: administrator/templates/isis/less/template.less [hunk 1]
```diff
@@ -212,19 +222,12 @@ body .navbar-fixed-top{
 }
 /* Header */
 .header{
-	#gradient > .vertical(@headerBackgroundHighlight, @headerBackground);
+	cackground: @headerBackground;
```
Comment (dgt41) [inline administrator/templates/isis/less/template.less]: background?

**Comment by dgt41** [comment]
:+1:

**Comment by nonumber** [comment]
This is a 5 minute improvement.
A new template... I don't see that happening at the moment.

**Comment by davdebcom** [comment]
@nonumber Great improvement. So subtle, but I like it already. Maybe the background colour on the login page can be lighter?

@brianteeman if nothing changes until a new template is ready, it's going to be a looong wait! :-P

**Comment by Fedik** [comment]
I liked more with gradient :ghost:

**Comment by brianteeman** [comment]
Thats the thing with design changes they are always subjective. Thats why i
suggest concentrating on something new

(Just look at the discussions on the sidebar icon)

On 6 March 2015 at 14:43, Fedir Zinchuk <notifications@github.com> wrote:

> I liked more with gradient [image: :ghost:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/6340#issuecomment-77569135>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by nonumber** [comment]
@davdebcom The background colour on the login page uses the main template colour setting. I find that having that dark (without the lighter gradient overlay) makes the form stand out more, which should be the focus of the page.

**Comment by nonumber** [comment]
So is there an active initiative for a new template? Is there a roadmap for it?

**Comment by n9iels** [comment]
It is planned for Joomla! 3.10, see: http://developer.joomla.org/cms/roadmap.html

I'm also for a new admin template, but I think we had to wait for the Bootstrap compatibility layer. Because building a new admin Template in Bootstrap 2 isn't a good idea.
Someone else made a PR (#4280) for a new Bootstrap 3 version of isis, see that PR for the made decision at that moment.

**Comment by nonumber** [comment]
Ok, so this has no chance. Dropped.

**Comment by dgt41** [comment]
@nonumber #4280 was changing the bootstrap files this one only touches template.less, so I think it might have a chance…
PS there are quite a few of us that we really dislike gradients

**Comment by zero-24** [comment]
I'm not sure who test this or only vote here. Can the tester mark your test into JTracker (http://issues.joomla.org/tracker/joomla-cms/6340)? So we can move this forward / RTC if we get two tests. Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6340">issues.joomla.org/joomla-cms/6340</a>.</sub>

**Comment by dgt41** [comment]
@test success
Besides the design subjective changes there is some improvement on the rendering speed of the pages. 
Here are the before:
Login
screen shot 2015-03-07 at 1 13 17
cpanel
screen shot 2015-03-07 at 1 12 54

After
Login
screen shot 2015-03-07 at 1 11 56
cpanel
screen shot 2015-03-07 at 1 12 27

Observe the rendering times and keep in mind that these are on a powerful Macbook Pro retina.
Bottom line: agree or disagree with the gradients we have to keep in mind that even the smallest touch in the codebase will reflect on the performance.

**Comment by nonumber** [comment]
A 12-15% rendering speed increase alone makes this worth it.

If it were the other way round, nobody in their right mind would choose to add a few gradients that would increase page rendering time by 10% of *every* page!

File: administrator/templates/isis/less/template.less [hunk 2]
```diff
@@ -1094,32 +1086,30 @@ a.grid_true {
 	border-top: 1px solid #d3d3d3;
 	border-bottom: 1px solid #eee;
 }
-.sidebar {
```
Comment (phproberto) [inline administrator/templates/isis/less/template.less]: Are you removing padding here intentionally?

Before:
padding

After:
no-padding

**Comment by phproberto** [comment]
Tested. It works here and I don't see why we cannot improve current template. Another gradient hater here.

So :+1: when the padding thing is answered.

**Comment by JoomliC** [comment]
<blockquote>
Thats the thing with design changes they are always subjective. Thats why i suggest concentrating on something new (Just look at the discussions on the sidebar icon)
</blockquote>
:+1: 
My opinion, as some of you, is to concentrate on functionnality improvement, and not design yet. The original reason for the sidebar was a "transition" before its future removal in core component (still some filters not migrated to search tools) (but to keep it for B/C) and yes, the icon debate was "juvénile"!

@nonumber About removal of gradients : +1 for me!
If it's loading the page faster, i would say let's go for it! As it's a performance improvement, and not a new design.

Thanks for your performance testing @dgt41 ! ;-)

So, i will be at jab, hope to meet some of you there (and see again a few of you met in Paris), and i would like, if not exist, to create a working group on a new admin template for J3 (@dgt41 yes, not all possible before 4.0, as related to your PR https://github.com/joomla/joomla-cms/pull/4280 ;-) ).

My proposal is to keep Isis as it is, EXCEPTED : 
- minor UX issue or improvement (full new design for a feature not accepted in Isis, but send to a repository for the new admin template)
- performance improvement (as for gradients)
- fix real bugs (wrong css declarations, issue of display on some device, mobile fixes....)

About a working group on a new admin template, it could be in the same time, a good way to write a "Joomla code of design UX/UI" for Joomla CMS (as there's one for Joomla coding standards, language style guide (new! and really useful!))

Cyril

**Comment by JoomliC** [comment]
@nonumber i would remove in the same time the box-shadow for class .view-login .well on login page, as with no gradient, the shadow is not really nice (and not needed, so less code, faster) ;-)
:+1:

**Comment by nonumber** [comment]
@JoomliC box shadows removed:

File: administrator/templates/isis/less/template.less (hunk 2 above)
Comment (nonumber) [inline administrator/templates/isis/less/template.less]: Yes, intentional.
Without the gradient, it looks broken with the margin there.
But forgot to fix the right margin. Fixed now:

**Comment by nonumber** [comment]
@dgt41 Can you check on your side if there is any performance improvements with the removed box-shadows and text-shadows?

File: administrator/templates/isis/less/template.less [hunk 3]
```diff
@@ -1094,32 +1084,30 @@ a.grid_true {
...
+	&.j-toggle-hidden {
+		right: -20px;
+	}
+	&.j-toggle-visible {
+		right: 10px;
+	}
 }
 .j-toggle-sidebar-button {
 	font-size: 14px;
 	color: #08c;
```
Comment (dgt41) [inline administrator/templates/isis/less/template.less]: Can you also remove the hard coded color here and put @linkColor

File: administrator/templates/isis/less/template.less [hunk 4]
```diff
@@ -1094,32 +1084,30 @@ a.grid_true {
...
 	text-decoration: none;
 	cursor: pointer;
+	&:hover {
+		color: #005580;
```
Comment (dgt41) [inline administrator/templates/isis/less/template.less]: Same here replace it with @linkColorHover

File: administrator/templates/isis/less/template.less (hunk 3 above)
Comment (nonumber) [inline administrator/templates/isis/less/template.less]: Those are outside the scope of this PR.
Please create a new PR to fix .less issues from the sidebar thing...

**Comment by davdebcom** [comment]
@test success
Works, no issues found. 

@nonumber from my testing:
Login before: 5.91s
Login after: 4.46s

Control Panel before: 4.79s
Control Panel after: 3.90s

File: administrator/templates/isis/less/template.less (hunk 3 above)
Comment (dgt41) [inline administrator/templates/isis/less/template.less]: :+1:

**Comment by nonumber** [comment]
So performance gain:
Login: ~25%
Control panel: ~19%

Not bad :)

File: administrator/templates/isis/less/template.less (hunk 4 above)
Comment (nonumber) [inline administrator/templates/isis/less/template.less]: Those are outside the scope of this PR.
Please create a new PR to fix .less issues from the sidebar thing...

**Comment by dgt41** [comment]
Please restore text shadow for .page-title as it breaks for white background
screen shot 2015-03-07 at 1 09 03

**Comment by nonumber** [comment]
@dgt41 Other solution for the light background colours implemented.
So now also improved for light template colours (top bar and login background).

File: administrator/templates/isis/less/template.less [hunk 5]
```diff
@@ -1070,22 +1077,18 @@ a.grid_true {
...
+	}
+	&.j-sidebar-visible {
+		left: 0;
+	}
+	.filter-select {
+		padding: 0 14px;
+	}
 }
 .j-toggle-sidebar-header {
 	color: #999;
```
Comment (JoomliC) [inline administrator/templates/isis/less/template.less]: Sorry for one more comment... ;-)
As the background is now greay on top (that's good) the "sidebar" header is not enought contrasted.
I propose this change : 
.j-toggle-sidebar-header {
	font-weight: bold;
	font-size: 14px;
	margin: -11px 0 7px 0;
	padding: 10px 0 10px 14px;
	border-top: 1px solid #d3d3d3;
	border-bottom: 1px solid #eee;
}

In the same time, is will fix this issue : https://github.com/joomla/joomla-cms/issues/6063

;-)

Comment (nonumber) [inline administrator/templates/isis/less/template.less]: Don't want to make this PR too big, but why is there a 'Sidebar' title there at all? Makes no sense! We don't put other titles in elements just telling it what it is, like 'Menu' and 'Header'!
I'd rather see:

Comment (brianteeman) [inline administrator/templates/isis/less/template.less]: LOL I just commented the same thing about this title not being needed on
the accessibility issue

On 7 March 2015 at 13:23, Peter van Westen <notifications@github.com> wrote:

> In administrator/templates/isis/less/template.less
> <https://github.com/joomla/joomla-cms/pull/6340#discussion_r25995964>:
>
> >  	border-right: 1px solid #d3d3d3;
> >  	border-bottom: 1px solid #d3d3d3;
> > -}
> > -.j-sidebar-container.j-sidebar-hidden {
> > -	left: -16.5%;
> > -}
> > -.j-sidebar-container.j-sidebar-visible {
> > -	left: 0;
> > +	&.j-sidebar-hidden {
> > +		left: -16.5%;
> > +	}
> > +	&.j-sidebar-visible {
> > +		left: 0;
> > +	}
> > +	.filter-select {
> > +		padding: 0 14px;
> > +	}
> >  }
> >  .j-toggle-sidebar-header {
> >  	color: #999;
>
> Don't want to make this PR too big, but why is there a 'Sidebar' title
> there at all? Makes no sense! We don't put other titles in elements just
> telling it what it is, like 'Menu' and 'Header'!
> I'd rather see:
>
> <https://camo.githubusercontent.com/25d669a256de45bca2e81adc1f5a95e32269f110/687474703a2f2f692e6e6f6e722e6e6c2f716c6367362e706e67>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/6340/files#r25995964>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

Comment (nonumber) [inline administrator/templates/isis/less/template.less]: :)

**Comment by davdebcom** [comment]
Nice!
Would it ba better to have jsidebar be over the entire height? I can imagine that it gives a more peaceful user interface when the sidebar does not change in height very single page/extension?

This:
screen shot 2015-03-07 at 15 15 31

Instead of the current situation:
screen shot 2015-03-07 at 15 15 58

**Comment by nonumber** [comment]
@davdebcom Not for now. Making it full height would require a lot of hoops to jump through. Unless we'd use flexboxes, but then you have an issue with 10% of the users (browser incompatibility).
So for now I'd say, just leave it.

**Comment by nonumber** [comment]
Should I add the removal of the 'Sidebar' title in this PR?

File: administrator/templates/isis/less/template.less (hunk 5 above)
Comment (JoomliC) [inline administrator/templates/isis/less/template.less]: I would say: let's go for it! 
Or admitted the fact that we will maybe able to remove the sidebar from core components in a few months, and that it will be kept only for B/C, maybe better to let the header title for now.
But for me, +1 for removal ;-)

**Comment by JoomliC** [comment]
@nonumber i think yes, this way, #6063 could be closed, and your PR will be ready for commit (as in my opinion, you have already done a good job!)
;-)

**Comment by JoomliC** [comment]
@davdebcom It's not possible to have a full height, as the sidebar layout is loaded in the component view, not the template. ;-) (we have already tried all possibilities before implemented the sidebar)

File: administrator/templates/isis/less/template.less (hunk 3 above)
Comment (nonumber) [inline administrator/templates/isis/less/template.less]: Ok, done in this PR now anyway :)

**Comment by n9iels** [comment]
sidebar

**Comment by nonumber** [comment]
Updated preview images in description (at the top).

All should be good now for final test runs.
Because of the changes made in last few commits, the test count should start over.

Major new changes (after last tests):
- Removed the title "Sidebar" from the sidebar
- Changed styling of sidebar a bit to accompany the title removal
- Added missing files (inverted images for light colours)
- Recompiled the css files
- Added minified css files for isis

**Comment by nonumber** [comment]
@n9iels Fixed.

**Comment by n9iels** [comment]
Thanks! verry good changes!
Little warning: When you're using com_patchtester and revert the patch you generate the error: Could not find template "isis".

**Comment by nonumber** [comment]
@n9iels Please report that to the patchtester code base. That is outside the scope/control of this PR.

Please test and report back here mentioning @test :)

**Comment by n9iels** [comment]
@test: Works fine now!

I'm going to report that, com_patchtester removed the whole index.php of issis for some reasons<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6340">issues.joomla.org/joomla-cms/6340</a>.</sub>

**Comment by nonumber** [comment]
@n9iels Thanks :)

**Comment by JoomliC** [comment]
@n9iels same issue with revert in com_patchtester, the index file was removed...

@nonumber Works well excepted an issue with extensions manager, the sidebar is not at top : 
capture d ecran 2015-03-07 a 18 02 45

**Comment by nonumber** [comment]
@JoomliC  Works fine for me. Browser cache?

**Comment by JoomliC** [comment]
@nonumber Still the same... seems to happen only when search tools is not there. @n9iels do you have tested with extensions manager ?

**Comment by nonumber** [comment]
Fine for me:

Is it possible for you to reproduce this online?

**Comment by tottello** [comment]
Personally I really like this gradients, especially on login form. Now login looks heavy, hard (like learning Joomla ;) ) Gradient on login was really nice, smooth and still fresh - this one should stay.

In admin template I don't care about gradients too much. Maybe I would do them just a little bit more flat, almost-flat but not flat). But even flat looks ok.
Second thing is border-top for header is to light. Opacity should be (IMO) around 0.1 to 0.15

Removing "Sidebar" title is good decision.
No gradient on sidebar is good decision.

**Comment by nonumber** [comment]
Oops, looks like I destroyed this PR...

**Comment by nonumber** [comment]
@tottello this is not only about removing gradients because it looks better (let's agree to disagree on what looks better), but also a performance gain.

**Comment by tottello** [comment]
IMO not worth it. This login page is less 'friendly'.

**Comment by gwsdesk** [comment]
@nonumber Peter I can reproduce on test site (online) same issue as JoomliC
extension-manager

Nice speed gain indeed!

**Comment by n9iels** [comment]
Same problem here. On some reasons the margin:0 0 18px; on form#adminForm is effective at the extensions manager.

**Comment by nonumber** [comment]
Ok, extension manager styling issue is fixed now. :)

**Comment by gwsdesk** [comment]
@test Confirmed. Looks to me all game, set and match



On 3/8/2015 4:37 PM, Peter van Westen wrote:
>
> Ok, extension manager styling issue is fixed now. :)
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/joomla/joomla-cms/pull/6340#issuecomment-77740962>.
>

-- 

Leo Lammerink
MD GWS - Enterprise Ltd
Skype: gwsgroup
www.gws-desk.com <http://www.gws-desk.com> | www.gws-host.com 
<http://www.gws-host.com> | www.gws-deals.today <http://gws-deals.today>

**Comment by n9iels** [comment]
@test sidebar in extensions manager works also fine for me

**Comment by dgt41** [comment]
@test success
I think this is ready for RTC

**Comment by zero-24** [comment]
Thanks! Moving to RTC!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6340">issues.joomla.org/joomla-cms/6340</a>.</sub>

**Comment by losedk** [comment]
Just a small tweak, could we add some padding to the logo?

padding

**Comment by nonumber** [comment]
I'll do that in another PR after this is merged. Otherwise testing needs to start over again.

**Comment by losedk** [comment]
@nonumber Great!

**Comment by JoomliC** [comment]
@nonumber @test success for me too!
Great job Peter! :+1:

**Comment by nonumber** [comment]
Thank you all for testing this and making it happen :)

**Comment by phproberto** [comment]
This cannot be merged anymore. @nonumber can you rebase it vs latest staging and fix conflicts please?

**Comment by nonumber** [comment]
All good now.

**Comment by phproberto** [comment]
It's merged. Thanks.

@nonumber  I had merged another PR before yours and to avoid asking you to fix conflicts (again) I merged it manually. I'd appreciate if you could check and confirm that the latest staging works as expected. Thanks!

**Comment by nonumber** [comment]
Looks good :)
Thanks

**Comment by phproberto** [comment]
Thanks for checking it

**Comment by jmccabe** [comment]
Well done guys; I've just spent around 1.5 hours trying to work out what I've installed that made the background all black on my login page because I didn't see any 'obvious user visible changes' listed on your changelog. Based on your performance improvement figures (0.8s per login was it?), that's about 6000 logins I have to do now to make up for the time lost.

**Comment by nonumber** [comment]
Thank you for your kind words!

**Comment by jmccabe** [comment]
You're welcome and, FWIW, I agree with @tottello; subjective as it may be, the login page looked far better with the gradients. The new layout is like something out of the '70s.

Once you've logged in the changes to the header are hardly noticeable so those aren't a big deal.

Relatively speaking how many times is the login page rendered compared to the other pages? Probably at least an order of magnitude less, so why worry about an occasional 0.5-0.8s and, given that the gradient was generated using inline styles from login.php, it had little effect on the amount of data being retrieved.

**Comment by dgt41** [comment]
@jmccabe I guess Bootstrap version 3.x which dropped totally the gradients is an old thing, right?
Also this one seems a ’70’s thing, because there are no gradients:
screen shot 2015-04-08 at 2 09 06
Check it live [here](https://cloud.digitalocean.com/login)
Let’s not make an argument on subjective matters! Can you mention something that this PR really made things worst for joomla? Accessibility, contrast, speed, bad rendering, anything we can start some discussion on a firm base?

**Comment by tottello** [comment]
@dgt41 Of course flat style could be done right. But instead of nice, smooth login page Joomla has heavy, dark login page now (your example is very light).

**Comment by dgt41** [comment]
@tottello changing it to something else is totally possible and quite easy:
screen shot 2015-04-08 at 4 18 01

```css
.view-login {background-color: #2089cf;}
```

**Comment by n9iels** [comment]
I'm also think that a bit lighter color looks better. Not so light as the examples, maybe someting like `#144576`?

@dgt41 if you have a good idea for a better color, please make a PR so we can discuss it farther over there :-)

**Comment by jmccabe** [comment]
@dgt41, subjectively speaking, yes, I do think that looks '70s, and I also think that you have to be careful what colour you choose; a lot of colours, especially blues, as a flat colour look 'cheap'. If you want to give an impression of quality you have to choose something that's fitting.

As for your question about thing being made worse for Joomla, in terms of its appearance, not a lot.

As I pointed out earlier, my main issue with this change is that it wasn't made obvious in any release notes that I saw that I should expect to see a difference when I next tried to login after updating. It's obviously in this list somewhere (https://github.com/joomla/joomla-cms/issues?page=10&q=&utf8=%E2%9C%93) but I'm not going to browse through 269 pages to look at every change before choosing to update. There's no requirement to re-login once an update to the main app has been done so I'd logged in with the gradient display, updated Joomla, then started installing the extensions I'd planned to do.

At some later point I noticed that the login page had changed so my immediate thought, as a user, is that something I'd installed had screwed things up. My first instinct was to uninstall the extensions since I assumed that, as Joomla! has masses of functionality that probably has loads of bugs that could do with being fixed, little effort would be getting expended on trivial things like changing how the admin login screen looks! Once uninstalling my recent extensions had made no difference it was then a case of trying to inspect elements of the view using Firefox, then trying to work out where that code was coming from, and trying to track down previous releases of Joomla! to download and compare (I knew 3.3.6 was fine as I have another site that I haven't updated to 3.4.1 yet), then try to find out what 'problem' meant this 'fix' had to be applied and, it turns out, the 'problem' is that some developer thinks gradients are old-hat!

**Comment by nonumber** [comment]
The background colour is connected to the ```Nav Bar Colour``` setting in the isis template settings.


We can easily change it to use the ```Header Colour ``` instead. That would result in:

**Comment by nonumber** [comment]
For instance:

**Comment by dgt41** [comment]
@nonumber Peter there is one more way to do it:
lets add custom.css also to login.php similar to index.php
https://github.com/joomla/joomla-cms/blob/staging/administrator/templates/isis/index.php#L27-L33
So people can totally customize to their own style the login page without another option on the backend

**Comment by jmccabe** [comment]
I can't see much point in going to the effort of changing it again (and annoying more people). It _would_ be nice if the templateColor field name in the code was renamed to be navbarColor though, more in keeping with the text displayed on the options shown above.

**Comment by jmccabe** [comment]
BTW - on a more positive note, the new icon and removal of the "sidebar" title on the sidebar are a definite improvement.

**Comment by nonumber** [comment]
See:  https://github.com/joomla/joomla-cms/pull/6703

**Comment by JoomliC** [comment]
The design, like fashion, are only loop process, where we always ended up back at a precise point already met. In stores, Disco fashion back! ;-)
